code/game/objects/__objs.dm

```diff
@@ -264,15 +267,21 @@
 	return TRUE
 
 /**
- * Init starting reagents and/or reagent var. Called if chem_volume > 0 in /obj/Initialize()
- * populate: If set to true, we expect map load/admin spawned reagents to be set.
+ * Init starting reagents and/or reagent var. Called in /obj/Initialize() if volume is above 0.
+ * Skips populate_initialize() if reagents is null, or if it is a list, ie. we are pending deserialization.
  */
-/obj/proc/initialize_reagents(var/populate = TRUE)
+/obj/proc/initialize_reagents()
 	SHOULD_CALL_PARENT(TRUE)
-	if(reagents?.total_volume > 0)
+	// Check if this is getting called twice, or we created reagents somewhere in Initialize() (bad juju)
+	if(istype(reagents))
 		log_warning("\The [src] possibly is initializing its reagents more than once!")
-	create_or_update_reagents(chem_volume)
-	if(populate)
+	// If preloaded from serde, handle expected list structure.
+	// Returns if preload is successful to skip populate_reagents() call.
+	FINALIZE_REAGENTS_SERDE_AND_RETURN(reagents)
```

Comment:
think occasionally `update_alpha_mask` on the fluid overlay can run before this does, which is a little concerning. led to runtimes locally